cli/azd/pkg/project/container_helper.go

```diff
@@ -216,6 +221,52 @@ func (ch *ContainerHelper) Deploy(
 ) *async.TaskWithProgress[*ServiceDeployResult, ServiceProgress] {
 	return async.RunTaskWithProgress(
 		func(task *async.TaskContextWithProgress[*ServiceDeployResult, ServiceProgress]) {
+			if dotnetDetails, ok := packageOutput.Details.(*dotNetSdkPublishConfiguration); ok {
```

Comment:
I think this set of changes can play well from your other PR with generic support for remote build.  Having all container build scenarios funnel through `ContainerHelper.Deploy` whether it is local build, generic remote build or dotnet publish will allow all service hosts to easily take advantage of these new features.